docs/src/howto-guides/021_enable_tedge_watchdog_using_systemd.md

```diff
@@ -0,0 +1,59 @@
+# Watchdog feature using `systemd` in `thin-edge.io`
+
+## Introduction
+
+`Watchdog` feature is used to check the health of a service/process by constantly exchanging
+the message between the watchdog process and the service/process that is being monitored.
+When the process does not update its health status within a specified `time` period, then the watchdog process will
+restart the service that is being monitored.
+
+This document shows how `thin-edge.io` services are managed using the systemd, then one can use `systemd`
+feature to check the health of these services as well. This document provides
+information about how to use `systemd` for checking the health of the services.
+
+## Enabling the `watchdog` feature in `systemd`
+
+Enabling the `watchdog` feature in systemd for a `thin-edge.io` service (tedge_agent, tedge_mapper_c8y/az/collectd)
+using the `systemd` is a two-step process.
+
+### Step 1: Enable the `watchdog` feature in the `systemd` service file
+For example to enable the `watchdog` feature for `tedge-mapper-c8y` service, update systemd service file as shown below.
+
+Add `tedge_watchdog.service` in  `After` under `[Unit]` section.
+Add `WatchdogSec=5` under `[Service]` section.
+
+The sample service file after updating looks as below.
+
+```shell
+[Unit]
+Description=tedge-mapper-c8y converts Thin Edge JSON measurements to Cumulocity JSON format.
+After=syslog.target network.target mosquitto.service tedge_watchdog.service
+
+[Service]
+User=tedge-mapper
+ExecStart=/usr/bin/tedge_mapper c8y
+Restart=on-failure
+RestartPreventExitStatus=255
+WatchdogSec=5
+```
+
+> Note: The systemd service file usually present in `/lib/systemd/system/tedge-mapper-c8y.service`.
+
+### Step 2: Start the `tedge-watchdog` service
+
```

Comment:
Better to add an intro of what the `tedge-watchdog` service is here, something like:
```suggestion

The `tedge-watchdog` service is responsible for periodically checking the health of all tedge services for which the watchdog feature is enabled, and send systemd watchdog notifications on their behalf to systemd.

```